spec/models/article_spec.rb

```diff
@@ -1091,6 +1091,33 @@ def foo():
     end
   end
 
+  describe "#user_mentions_in_markdown" do
+    before do
+      stub_const("Article::MAX_USER_MENTIONS", 7)
+      stub_const("Article::MAX_USER_MENTION_LIVE_AT", 1.day.ago) # Set live_at date to a time in the past
+    end
+
+    it "is valid with any number of mentions if created before MAX_USER_MENTION_LIVE_AT date" do
+      # Explicitly set created_at date to a time before MAX_USER_MENTION_LIVE_AT
+      article = create(:article, created_at: 3.days.ago)
+
+      article.body_markdown = "hi @#{user.username}! " * 8
+      expect(article).to be_valid
+    end
+
+    it "is valid with seven or fewer mentions if created after MAX_USER_MENTION_LIVE_AT date" do
+      article.body_markdown = "hi @#{user.username}! " * 7
```

Comment:
To make these specs more robust you could use `Article::MAX_USER_MENTIONS` and `Article::MAX_USER_MENTIONS + 1` above when duplicating the mentions, that way if you change the constant none of the hardcoded numbers in the specs will need to be updated

Reply:
Oooooh of course! Great idea, love it. [This latest commit](https://github.com/forem/forem/pull/13259/commits/c1d6b111b426d4517d0692086a4e5ae57d975bcc) takes that into account 👍🏽